pkg/cri/resource-manager/policy/builtin/memtier/pools.go

```diff
@@ -391,13 +395,8 @@ func (p *policy) releasePool(container cache.Container) (Grant, bool, error) {
 
 	log.Debug("  => releasing grant %s...", grant)
 
-	pool := grant.GetCPUNode()
-	supply := pool.FreeSupply()
-	supply.ReleaseCPU(grant)
-
-	pool = grant.GetMemoryNode()
-	supply = pool.FreeSupply()
-	supply.ReleaseMemory(grant)
+	grant.GetCPUNode().FreeSupply().ReleaseCPU(grant)
+	grant.GetMemoryNode().FreeSupply().ReleaseMemory(grant)
```

Comment:
nitpick: I'd also probably have a `Supply.Release()` defined and used here. Both for symmetry and as the preferred function to be used normally for releasing a grant fully. `ReleaseMemory()` would then be reserved for the special cases like the 'fixup' algorithm.

Reply:
Hmm... the reason it's done that way is because the CPU and memory can nowadays be allocated from a different `Supply`. We could have the `Release()´ function to be for `Grant` and not `Supply` though.